powerauth-push-server/src/main/java/io/getlime/push/service/PushMessageSenderService.java

```diff
@@ -254,7 +236,7 @@ public void sendCampaignMessage(Long appId, String platform, String token, PushM
      * @throws PushServerException In case any issue happens while sending the push message. Detailed information about
      * the error can be found in exception message.
      */
-    public void sendCampaignMessage(final Long appId, String platform, final String token, PushMessageBody pushMessageBody, PushMessageAttributes attributes, String userId, Long deviceId, String activationId) throws PushServerException {
+    private void sendCampaignMessage(final Long appId, String platform, final String token, PushMessageBody pushMessageBody, PushMessageAttributes attributes, String userId, Long deviceId, String activationId) throws PushServerException {
```

Comment:
This method is supposed to be public. While the compiler probably complains with an info message that "The access can be private" (since this method is not used at the moment with controller code), this is an Java code API method extending the `sendCampaignMessage()` method that is just above with possibility to use `attributes`.